### PR TITLE
APIGW: merge (disabled) APIGW branch

### PR DESCRIPTION
**Pull request by justinmk3**
## Description

Merge disabled APIGW code to main branch, to avoid conflicts with OCI work.

## Review Notes

- Only the last 2 commits need review. All other commits are a clean merge from `feature/apigateway`.
- No `package.json` changes because the APIGW items are not  global, they only apply to AWS Explorer context-menu.

## Testing

1. "API" codelenses do not appear in `template.yaml`
2. "API Gateway" node does not appear in AWS Explorer

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by hunterwerlla — APPROVED** [review]

File: src/shared/featureToggle.ts [hunk 1]
```diff
@@ -89,6 +90,9 @@ export class FeatureToggle {
      * @param key Feature key to search for
      */
     public isFeatureActive(key: string): boolean {
+        if (key === ActiveFeatureKeys.ApiGateway) {
+            return false
```
Comment (justinmk3) [inline src/shared/featureToggle.ts:94]: always false

File: src/awsexplorer/regionNode.ts [hunk 2]
```diff
@@ -38,12 +41,17 @@ export class RegionNode extends AWSTreeNodeBase {
...
 
         //  Services that are candidates to add to the region explorer.
         //  `serviceId`s are checked against ~/resources/endpoints.json to see whether or not the service is available in the given region.
         //  If the service is available, we use the `createFn` to generate the node for the region.
         //  This interface exists so we can add additional nodes to the array (otherwise Typescript types the array to what's already in the array at creation)
+        const partitionId = regionProvider.getPartitionId(this.regionCode) ?? DEFAULT_PARTITION
         const serviceCandidates = [
+            ...(!isApigwEnabled
+                ? []
+                : [{ serviceId: 'apigateway', createFn: () => new ApiGatewayNode(partitionId, this.regionCode) }]),
```
Comment (justinmk3) [inline src/awsexplorer/regionNode.ts:54]: disabled

File: src/shared/cloudformation/templateSymbolResolver.ts [hunk 3]
```diff
@@ -27,44 +24,110 @@ export interface TemplateSymbolProvider {
...
+        const allSymbols = await this.symbolProvider.getSymbols(this.document, waitForSymbols)
+        // Only want top level (TODO: is this actually true?)
+        const funSymbols = (
+            allSymbols.find(o => o.name === 'Resources' && o.kind === vscode.SymbolKind.Module)?.children ?? []
+        ).filter(r => this.isCfnType(CloudFormation.SERVERLESS_FUNCTION_TYPE, r))
+        if (kind === 'function') {
+            return funSymbols.map(r => ({ name: r.name, range: r.range, kind: kind }))
+        }
+        if (!isApigwEnabled) {
+            return []
```
Comment (justinmk3) [inline src/shared/cloudformation/templateSymbolResolver.ts:52]: disabled

**Pull request merged**